### ISSUE TITLE
exception when operating PyWren

### ISSUE THREAD

**Issue by omerb01** [pywren/pywren-ibm-cloud#228]
all tests fail on current master branch, for example:
```
python -m pywren_ibm_cloud.tests -t test_call_async
```
produces:
```
Uploading test files...
Testing call_async()...
PyWren v1.2.1-SNAPSHOT init for IBM Cloud Functions - Namespace: pywren_dev_us_east - Region: us_east
ExecutorID 2360-4ee0 | JobID A000 - Selected Runtime: ibmfunctions/action-python-v3.6 - 256MB (Installing...)
ExecutorID 2360-4ee0 | JobID A000 - Uploading function and data - Total: 656.0B
ExecutorID 2360-4ee0 | JobID A000 - Starting function invocation: hello_world()  - Total: 1 activations
E
======================================================================
ERROR: test_call_async (__main__.TestPywren)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/omerbelhasin/IdeaProjects/pywren-ibm-cloud/pywren_ibm_cloud/tests.py", line 209, in test_call_async
    pw.call_async(hello_world, "")
  File "/Users/omerbelhasin/IdeaProjects/pywren-ibm-cloud/pywren_ibm_cloud/executor.py", line 143, in call_async
    future = self.invoker.run(job)
  File "/Users/omerbelhasin/IdeaProjects/pywren-ibm-cloud/pywren_ibm_cloud/invoker.py", line 196, in run
    if self.failed_calls_queue.qsize() > 0:
  File "/Users/omerbelhasin/.pyenv/versions/3.6.7/lib/python3.6/multiprocessing/queues.py", line 117, in qsize
    return self._maxsize - self._sem._semlock._get_value()
NotImplementedError

----------------------------------------------------------------------
Ran 1 test in 5.701s

FAILED (errors=1)
Deleting test files...
```

**Comment by gilv** [comment]
@JosepSampe this is something in your latest pull request. I have the same issue

**Comment by JosepSampe** [comment]
Yep, fixing it..

**Comment by JosepSampe** [comment]
Fixed in #229 
I temporally deactivated **test_internal_executions**

**Comment by JosepSampe** [comment]
I already fixed **test_internal_executions** test, i.e. pywren inside pywren. Will include it in the next PR

**Comment by omerb01** [comment]
@JosepSampe I submited a PR with more tests that have not passed yet and I also reverted `test_internal_executions`, lets keep all tests active even if they currently aren't passing

**Comment by omerb01** [comment]
@JosepSampe all tests work except `test_multiple_executions`

when I do:
```
python -m pywren_ibm_cloud.tests -t test_multiple_executions
```
I get:
```
Uploading test files...
Testing multiple executions...
PyWren v1.2.1-SNAPSHOT init for IBM Cloud Functions - Namespace: pywren_dev_us_east - Region: us_east
ExecutorID c2a0ec/0 | JobID M000 - Selected Runtime: ibmfunctions/action-python-v3.6 - 256MB 
ExecutorID c2a0ec/0 | JobID M000 - Uploading function and data - Total: 715.0B
ExecutorID c2a0ec/0 | JobID M000 - Starting function invocation: simple_map_function()  - Total: 2 activations
ExecutorID c2a0ec/0 | JobID M001 - Selected Runtime: ibmfunctions/action-python-v3.6 - 256MB 
ExecutorID c2a0ec/0 | JobID M001 - Uploading function and data - Total: 715.0B
ExecutorID c2a0ec/0 | JobID M001 - Starting function invocation: simple_map_function()  - Total: 2 activations
ExecutorID c2a0ec/0 - Getting results...

  100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4  

ExecutorID c2a0ec/0 - Cleaning temporary data
ExecutorID c2a0ec/0 | JobID M002 - Selected Runtime: ibmfunctions/action-python-v3.6 - 256MB 
ExecutorID c2a0ec/0 | JobID M002 - Uploading function and data - Total: 715.0B
ExecutorID c2a0ec/0 | JobID M002 - Starting function invocation: simple_map_function()  - Total: 2 activations
ExecutorID c2a0ec/0 - Getting results...

    0%|                                                                                                                                                                                                                                                           | 0/2  

ExecutorID c2a0ec/0 | JobID M002 - There was an exception - Activation ID: fe3fb5a3cf8e449cbfb5a3cf8ef49cd5

Traceback (most recent call last):
  File "/action/pywren_ibm_cloud/runtime/function_handler/jobrunner.py", line 237, in run
  File "/action/pywren_ibm_cloud/runtime/function_handler/jobrunner.py", line 89, in _get_function_and_modules
  File "/action/pywren_ibm_cloud/storage/storage.py", line 101, in get_func
  File "/action/pywren_ibm_cloud/storage/backends/ibm_cos/ibm_cos.py", line 154, in get_object
Exception: No such key /omeruseast/pywren.jobs/c2a0ec/0/M002/func.pickle found in storage.

ExecutorID c2a0ec/0 - Cleaning temporary data
E
======================================================================
ERROR: test_multiple_executions (__main__.TestPywren)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/omerbelhasin/IdeaProjects/pywren-ibm-cloud/pywren_ibm_cloud/executor.py", line 367, in wait
    THREADPOOL_SIZE=THREADPOOL_SIZE, WAIT_DUR_SEC=WAIT_DUR_SEC)
  File "/Users/omerbelhasin/IdeaProjects/pywren-ibm-cloud/pywren_ibm_cloud/wait/wait_storage.py", line 72, in wait_storage
    THREADPOOL_SIZE=THREADPOOL_SIZE)
  File "/Users/omerbelhasin/IdeaProjects/pywren-ibm-cloud/pywren_ibm_cloud/wait/wait_storage.py", line 229, in _wait_storage
    pool.map(get_result, f_to_wait_on)
  File "/Users/omerbelhasin/.pyenv/versions/3.6.7/lib/python3.6/multiprocessing/pool.py", line 288, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/Users/omerbelhasin/.pyenv/versions/3.6.7/lib/python3.6/multiprocessing/pool.py", line 670, in get
    raise self._value
  File "/Users/omerbelhasin/.pyenv/versions/3.6.7/lib/python3.6/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/Users/omerbelhasin/.pyenv/versions/3.6.7/lib/python3.6/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/Users/omerbelhasin/IdeaProjects/pywren-ibm-cloud/pywren_ibm_cloud/wait/wait_storage.py", line 213, in get_result
    f.result(throw_except=throw_except, internal_storage=internal_storage)
  File "/Users/omerbelhasin/IdeaProjects/pywren-ibm-cloud/pywren_ibm_cloud/future.py", line 220, in result
    self.status(throw_except=throw_except, internal_storage=internal_storage)
  File "/Users/omerbelhasin/IdeaProjects/pywren-ibm-cloud/pywren_ibm_cloud/future.py", line 171, in status
    raise FunctionException(self.executor_id, self.job_id, self.activation_id, self._exception, msg)
pywren_ibm_cloud.future.FunctionException: ExecutorID c2a0ec/0 | JobID M002 - There was an exception - Activation ID: fe3fb5a3cf8e449cbfb5a3cf8ef49cd5

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/omerbelhasin/IdeaProjects/pywren-ibm-cloud/pywren_ibm_cloud/tests.py", line 279, in test_multiple_executions
    result = pw.get_result()
  File "/Users/omerbelhasin/IdeaProjects/pywren-ibm-cloud/pywren_ibm_cloud/executor.py", line 457, in get_result
    WAIT_DUR_SEC=WAIT_DUR_SEC)
  File "/Users/omerbelhasin/IdeaProjects/pywren-ibm-cloud/pywren_ibm_cloud/executor.py", line 387, in wait
    sys.exit()
SystemExit

----------------------------------------------------------------------
Ran 1 test in 6.759s

FAILED (errors=1)
Deleting test files...
```

**Comment by JosepSampe** [comment]
@omerb01 Do you have `data_cleaner : True` in config file?

**Comment by omerb01** [comment]
@JosepSampe yep

**Comment by JosepSampe** [comment]
@omerb01 Yep, I already found the issue, will submit in next PR ;)

**Comment by JosepSampe** [comment]
@omerb01 Running some tests I can see that the results file from tests (`__pywren.test/result`) is always kept in storage. Is this the correct behavior? or it should be deleted at the end of the tests?

**Comment by JosepSampe** [comment]
@omerb01 @gilv Submitted fix in #233

**Comment by omerb01** [comment]
>@omerb01 Running some tests I can see that the results file from tests (__pywren.test/result) is always kept in storage. Is this the correct behavior? or it should be deleted at the end of the tests?

@JosepSampe changed behavior on #234
